TypeSystem/subtyping/test_types/extension_type_A11.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a type T0 is a subtype of a type T1 (written
+/// T0 <: T1) when:
+/// Assume that DV declares an extension type declaration named Name with type
+/// parameters X1 .. Xs, and V1 is a superinterface of DV. Then Name<T1, .. Ts>
+/// is a subtype of [T1/X1 .. Ts/Xs]V1 for all T1, .. Ts.
+///
+/// @description Check that an extension type `ET<T1, ..., Ts>` is a subtype of
+/// an extension type `ET<X1, ..., Xs>` if `Ti` is a subtype of `Xi` for all
+/// `i`. Test union types.
```

Comment:
Wouldn't something like this be a better fit?
```suggestion
/// @description Check that an extension type `ET` that implements `S` is a
/// subtype of `FutureOr<S>`.
```